lib/tests/streamlit/runtime/app_session_test.py

```diff
@@ -1128,6 +1183,29 @@ def test_can_specify_all_options(self, patched_config):
             ),
         ]
 
+        app_session._populate_theme_msg(
+            new_session_msg.custom_theme.sidebar,
+            "theme.sidebar",
+        )
+        assert new_session_msg.custom_theme.HasField("sidebar")
+        assert new_session_msg.custom_theme.sidebar.primary_color == "red"
+        assert new_session_msg.custom_theme.sidebar.background_color == "white"
+        assert new_session_msg.custom_theme.sidebar.text_color == "black"
+        assert new_session_msg.custom_theme.sidebar.secondary_background_color == "blue"
+        assert new_session_msg.custom_theme.sidebar.base_radius == "1.2rem"
+        assert new_session_msg.custom_theme.sidebar.border_color == "#ff0000"
+        assert new_session_msg.custom_theme.sidebar.show_border_around_inputs is True
+        assert new_session_msg.custom_theme.sidebar.link_color == "#2EC163"
+        assert new_session_msg.custom_theme.sidebar.heading_font == "Inter Bold"
+        assert new_session_msg.custom_theme.sidebar.body_font == "Inter"
+        assert new_session_msg.custom_theme.sidebar.code_font == "Monaspace Argon"
+
+        # Default values for unsupported fields in sidebar
+        assert new_session_msg.custom_theme.sidebar.base == 0
+        assert new_session_msg.custom_theme.sidebar.base_font_size == 0
+        assert not new_session_msg.custom_theme.sidebar.font_faces
+        assert new_session_msg.custom_theme.sidebar.show_sidebar_separator is False
```

Comment:
I think for the props that have an `optional` keyword, you should also be able to check that the field wasn't set via:

```
assert not new_session_msg.custom_theme.sidebar.HasField("base_font_size")
```